libs/commons/List_.ml

```diff
@@ -325,6 +325,11 @@ let optlist_to_list = function
 
 let sort xs = List.sort compare xs
 
+let sort_by_key key cmp xs =
```

Comment:
maybe I would rename sort_by_mapped_key,
to avoid ambiguity with the Assoc_.sort_by_key_xxx function
and I would rename key as fkey, so it's clear we're passing a function here, not a key.

Reply:
>  I would rename key as fkey, so it's clear we're passing a function here, not a key.

I am not certain what would be unclear here, given the function is 3 lines and the one usage of this is that it is applied. I can rename it, but I think this probably speaks more to the fact that this warrants documentation. I will add some in the mli.

> maybe I would rename sort_by_mapped_key,
to avoid ambiguity with the Assoc_.sort_by_key_xxx function

I would rather not rename this, since I think the suggestion simply increases verbosity and imo any ambiguity is resolved by the module name.